### PR TITLE
Add tests for package model/flashcard

### LINKED ISSUE #88: Create Test Classes
For:
- OpenEndedQuestion
- MultipleChoiceQuestion
- Option
- Question
- QuestionContainingKeywordPredicate
- FlashcardContainTagPredicate
- Statistics
- UniqueFlashcardList

### PR DESCRIPTION
**Pull request by josiahkhoo**
Closes #88

### CODE REVIEW

File: src/test/java/quickcache/model/flashcard/ChoiceTest.java [hunk 1]
```diff
@@ -35,4 +35,24 @@ public void isValidChoice() {
...
+        // same object -> returns true
+        assertTrue(choice.equals(choice));
+
+        // null -> returns false
+        assertFalse(choice.equals(null));
+
+        // different type -> returns false
+        assertFalse(choice.equals(5));
+
+        // different person -> returns false
```
Comment (FH-30) [inline src/test/java/quickcache/model/flashcard/ChoiceTest.java]: Might need to rename person to flashcard

File: src/test/java/quickcache/model/flashcard/QuestionContainsKeywordsPredicateTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            new QuestionContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different person -> returns false
```
Comment (FH-30) [inline src/test/java/quickcache/model/flashcard/QuestionContainsKeywordsPredicateTest.java]: Same here.

File: src/test/java/quickcache/model/flashcard/OptionTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        // same object -> returns true
+        assertTrue(option.equals(option));
+
+        // null -> returns false
+        assertFalse(option.equals(null));
+
+        // different type -> returns false
+        assertFalse(option.equals(5));
+
+        // different person -> returns false
```
Comment (FH-30) [inline src/test/java/quickcache/model/flashcard/OptionTest.java]: Same here.

File: src/test/java/quickcache/model/flashcard/FlashcardContainsTagPredicateTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        FlashcardContainsTagPredicate firstPredicateCopy = new FlashcardContainsTagPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different person -> returns false
```
Comment (FH-30) [inline src/test/java/quickcache/model/flashcard/FlashcardContainsTagPredicateTest.java]: Same here.

**Review by FH-30 — APPROVED** [review]
LGTM! Thanks for helping with this. Just a few nits to fix.

File: src/test/java/quickcache/model/flashcard/ChoiceTest.java (hunk 1 above)
Comment (josiahkhoo) [reply src/test/java/quickcache/model/flashcard/ChoiceTest.java]: Fixed in 1d082d2

File: src/test/java/quickcache/model/flashcard/QuestionContainsKeywordsPredicateTest.java (hunk 2 above)
Comment (josiahkhoo) [reply src/test/java/quickcache/model/flashcard/QuestionContainsKeywordsPredicateTest.java]: Fixed in 1d082d2

File: src/test/java/quickcache/model/flashcard/OptionTest.java (hunk 3 above)
Comment (josiahkhoo) [reply src/test/java/quickcache/model/flashcard/OptionTest.java]: Fixed in 1d082d2

File: src/test/java/quickcache/model/flashcard/FlashcardContainsTagPredicateTest.java (hunk 4 above)
Comment (josiahkhoo) [reply src/test/java/quickcache/model/flashcard/FlashcardContainsTagPredicateTest.java]: Fixed in 1d082d2

**Pull request merged**